src/databricks/labs/dqx/installer/config_provider.py

```diff
@@ -319,3 +331,65 @@ def _prompt_quarantine_config(self, is_streaming: bool) -> OutputConfig | None:
                 trigger=quarantine_trigger_options,
             )
         return None
+
+    def _prompt_metrics_config(self, is_streaming: bool) -> OutputConfig:
+        """Prompt user for metrics configuration."""
+        metrics_table = self._prompts.question(
+            "Provide table for storing summary metrics in the fully qualified format `catalog.schema.table` or `schema.table`",
+            valid_regex=r"^([\w]+(?:\.[\w]+){1,2})$",
+        )
+
+        metrics_write_mode = self._prompts.question(
+            "Provide write mode for metrics table (e.g. 'append' or 'overwrite')",
+            default="append",
+            valid_regex=r"^(append|overwrite)$",
+        )
+
+        metrics_format = self._prompts.question(
+            "Provide format for the metrics data (e.g. delta, parquet)",
+            default="delta",
+            valid_regex=r"^\w.+$",
+        )
+
+        metrics_write_options = json.loads(
+            self._prompts.question(
+                "Provide additional options for writing the metrics data (e.g. {\"mergeSchema\": \"true\"})",
+                default="{}",
+                valid_regex=r"^.*$",
+            )
+        )
+
+        metrics_trigger_options = {}
+        if is_streaming:
+            metrics_trigger_options = json.loads(
+                self._prompts.question(
+                    "Provide additional options for writing the metrics data using streaming "
+                    "(e.g. {\"availableNow\": true})",
+                    default="{}",
+                    valid_regex=r"^.*$",
+                )
+            )
+
+        return OutputConfig(
+            location=metrics_table,
+            mode=metrics_write_mode,
+            format=metrics_format,
+            options=metrics_write_options,
+            trigger=metrics_trigger_options,
+        )
+
+    def _prompt_custom_metrics(self) -> list[str]:
+        """Prompt user for custom metrics as Spark SQL expressions."""
+        custom_metrics_input = self._prompts.question(
```

Comment:
Maybe bettrer if we require to provide this as a list: ["count(case when age > 65 then 1 end) as senior_count", "avg(salary) as avg_salary"]
You can just use json.loads(s) to load it